### PR TITLE
Feature/add mapping asset credential

### PR DESCRIPTION
**Pull request by perrettecl**
Implement:
- Create a mapping
- Read a mapping
- Update a mapping: Credential, Status of credential, Extended Info
- Delete a mapping

### CODE REVIEW

**Review by FrancoisRegisDegott-eaton — APPROVED** [review]

File: include/cam_exception.h [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+namespace cam
+{
+    enum ErrorCode : uint8_t
+    {
+        GENERIC = 0,
+        UNSUPPORTED_COMMAND,
+        PROTOCOL_ERROR,
+        BAD_COMMAND_ARGUMENT,
+        MAPPING_DO_NOT_EXIST,
```
Comment (boricj) [inline include/cam_exception.h]: Correct to `MAPPING_DOES_NOT_EXIST`.

File: include/cam_exception.h [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        MAPPING_ALREADY_EXIST,
```
Comment (boricj) [inline include/cam_exception.h]: Correct to `MAPPING_ALREADY_EXISTS`.

File: include/cam_exception.h [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+            CamException(whatArg, ErrorCode::BAD_COMMAND_ARGUMENT)
+        {
+            extraData.getMember("argument") >>= m_argument;
+        }
+
+        inline std::string getArgument() const { return m_argument; }
+    };
+    
+// Mapping do not exist
+    class CamMappingDoNotExistException : public CamException
```
Comment (boricj) [inline include/cam_exception.h]: Correct to `CamMappingDoesNotExistException`.

File: include/cam_exception.h [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+            extraData.getMember("asset") >>= m_asset;
+            extraData.getMember("usage") >>= m_usage;
+        }
+
+        inline std::string getAssetId() const { return m_asset; }
+        inline std::string getUsageId() const { return m_usage; }
+    };
+
+// Mapping already exist
+    class CamMappingAlreadyExistException : public CamException
```
Comment (boricj) [inline include/cam_exception.h]: Correct to `CamMappingAlreadyExistsException`.

**Review by boricj — COMMENTED** [review]
I'm starting to think we're building a full-blown NoSQL database in disguise. Should we consider putting all that information inside `box_utf8` and take advantage of access rights and table encryption that MySQL already provides instead of reinventing everything?

**Pull request merged**